ecl/hql/hqlmeta.cpp

```diff
@@ -1890,6 +1904,23 @@ IHqlExpression * preserveTableInfo(IHqlExpression * newTable, IHqlExpression * o
 
 //---------------------------------------------------------------------------------------------------------------------
 
+bool hasOrderedAttribute(IHqlExpression * expr)
+{
+    return expr->hasAttribute(orderedAtom) || expr->hasAttribute(_ordered_Atom);
+}
+
+bool isOrdered(IHqlExpression * expr)
+{
+    if (expr->hasAttribute(orderedAtom))
+        return getBoolAttribute(expr, orderedAtom, true);
+    return getBoolAttribute(expr, _ordered_Atom, true);
```

Comment:
yes.  By default the order of ecl operations is specified, unless explicitly indicated otherwise.